src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -1188,7 +1183,7 @@ - (void)unitTestInjectAttributeReport:(NSArray<NSDictionary<NSString *, id> *> *
 {
     dispatch_async(self.queue, ^{
         [self _handleReportBegin];
-        [self _handleAttributeReport:attributeReport];
+        [self _handleAttributeReport:attributeReport fromSubscription:NO];
```

Comment:
The unit test injection is supposed to look like a subscription, so we should pass YES here, imo.  Or make it configurable by caller.

Reply:
Made `unitTestInjectAttributeReport` configurable by caller and set the two current callsites to pass `fromSubscription:YES`.